From: Sergio Durigan Junior <sergiodj@sergiodj.net>
Date: Sun, 27 Dec 2015 01:34:55 -0500
Subject: Improve security by making sure that we use a safe set of ciphers.

This comes from Debian bug #804196 (LP Bug #1517265).  When Midori
uses libwebkitgtk, it inherits the default value of the
G_TLS_GNUTLS_PRIORITY environment variable.  Unfortunately this value
is not safe enough because it still lets GnuTLS to enable to unsafe
ciphers.  Therefore, we need to set this variable ourselves and make
sure that Midori is using safe ciphers for TLS operations.
---
 midori/main.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/midori/main.c b/midori/main.c
index 00bc3b4..8954e42 100644
--- a/midori/main.c
+++ b/midori/main.c
@@ -143,6 +143,11 @@ main (int    argc,
     /* Versioned prgname to override menuproxy blacklist */
     g_set_prgname (PACKAGE_NAME "4");
 
+    /* Make sure we are using a sane set of ciphers.  */
+    g_setenv ("G_TLS_GNUTLS_PRIORITY",
+	      "NORMAL:%COMPAT:%LATEST_RECORD_VERSION:!VERS-SSL3.0:!ARCFOUR-128",
+	      1);
+
     if (version)
     {
         GString* versions = g_string_new ("");
